src/string_cache.rs

```diff
@@ -1,43 +1,46 @@
-use crate::binxml::name::BinXmlName;
-use crate::err::{DeserializationResult, WrappedIoError};
+use crate::binxml::name::{BinXmlName, BinXmlNameLink};
+use crate::err::DeserializationResult;
 use crate::ChunkOffset;
 
+use log::trace;
+use std::borrow::BorrowMut;
 use std::collections::HashMap;
 use std::io::{Cursor, Seek, SeekFrom};
 
-pub type StringHash = u16;
+#[derive(Debug)]
+pub struct StringCache<'a>(HashMap<ChunkOffset, BinXmlName<'a>>);
```

Comment:
Can the cache contain not-owned strings? Maybe we should simplify here with `HashMap<ChunkOffset, String>`?

Reply:
Done.